src/hx/libs/ssl/Build.xml

```diff
@@ -2,125 +2,143 @@
 
 <pragma once="true" />
 
-<set name="MBEDTLS_DIR" value="${HXCPP}/project/thirdparty/mbedtls-2.28.2" />
+<set name="MBEDTLS_DIR" value="${HXCPP}/project/thirdparty/mbedtls-3.6.0" />
 
 <files id="hxcpp_ssl" >
 	<depend files="hxcpp-depends"/>
 	<depend name="${this_dir}/Build.xml" dateOnly="true" />
 	<cache value="true" asLibrary="true" />
+    <compilerflag value="-std=c99" unless="MSVC_VER"/>
 	<compilerflag value="-I${MBEDTLS_DIR}/include"/>
-	<compilerflag value="-I${this_dir}"/>
-	<compilerflag value="-std=c99" unless="MSVC_VER" />
+    <compilerflag value="-I${this_dir}"/>
+    <compilerflag value="-DWIN32_LEAN_AND_MEAN" if="windows"/>
```

Comment:
I ran into this issue when porting neko as well. It may make more sense to put this definition directly in `threading_alt.h`, where it is needed:
https://github.com/HaxeFoundation/neko/commit/31e9db63291476e0aa811653e5518db9abe77b2f#diff-c5bad8f9ad3067b73cb76e84d84c5fef6f46ad66600fa9f8356ecfc0240158fd

Reply:
Shifting the lean and mean define to threading_alt for hxcpp doesn't do the trick. You'll still get redefinition errors.

Reply:
I tried this patch on my machine and hxcpp_ssl was compiling fine: https://github.com/tobil4sk/hxcpp/commit/a51f140c7f3223fdf6d4ae5b1d5ecaef38f7e143. It's not breaking the windows ci either: https://github.com/tobil4sk/hxcpp/actions/runs/9809645213/job/27088061829

The reason I would prefer to have it in `threading_alt.h` is that reducing the number of command line flags required to compile hxcpp's mbedtls makes it easier for lime to link against it, which would solve lime's mbedtls issues.

Reply:
Just gave it another try and it works now. I can't remember where the error I was getting was coming from since I tried moving the lean and mean define around the place pretty early on in this branch. Probably some compiler cache invalidation issue.